knip.config.ts

```diff
@@ -0,0 +1,49 @@
+import { KnipConfig } from 'knip';
+
+type UnwrapConfig<T> = T extends (...args: any[]) => infer R // 1. Is it a function? Get the return type.
+  ? UnwrapConfig<R> // 2. Recurse (in case it's () => Promise<config>)
+  : T extends Promise<infer P> // 3. Is it a Promise? Get the resolved type.
+    ? UnwrapConfig<P> // 4. Recurse
+    : T;
+
+type packageConfig = Exclude<UnwrapConfig<KnipConfig>['workspaces'], undefined>[string];
+
+const basePackageConfig: packageConfig = {
+  entry: ['src/index.ts'],
+  vitest: { config: 'vitest.config.cts', entry: ['tests/**/*.spec.ts'] },
+  typescript: {
+    config: ['tsconfig.json', 'tsconfig.build.json'],
+  },
+};
```

Comment:
```suggestion
} as const;
```

Reply:
why? i dont care about the type afterwards